### PR TITLE
add link payment method for capture later option

### LINKED ISSUE #4821: StripeLink: Link is not enabled on frontend for 'capture later'
### To Reproduce

1. Go to WC Admin
2. Enable Link payment method
3. Enable 'Capture later'
4. Link is not active on frontend

### Actual behavior

Link payment method is not kicking in when 'capture later' flag is enabled.

### Expected behavior

Link should work also when capture later is enabled.

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #4821 

#### Changes proposed in this Pull Request
At this point Link payment method is not enabled when capture later flag is on. Since payments with Link can also be captured later, we should enable Link also when capture later flag is on.

#### Testing instructions

* Enable Link payment method
* Set 'Capture later' flag to on. (`WC -> Settings -> Payments-> Transactions->Issue an authorization on checkout, and capture later`) 
* create an order and pay it with Link
* no errors should occur.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: changelog/fix-4821-link-enabled-capture-later [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Add link payment method for capture later option
```
Comment (kalessil) [inline changelog/fix-4821-link-enabled-capture-later]: Can we tweak the wording? 

As I read, the change is about enabling Link to support authorization/capture scenarios.
And the changelog entry is misleading from this angle.

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 2]
```diff
@@ -924,7 +924,7 @@ function() use ( $payment_fields ) {
 	 */
 	public function get_payment_method_ids_enabled_at_checkout( $order_id = null, $force_currency_check = false ) {
 		$capture                    = empty( $this->get_option( 'manual_capture' ) ) || $this->get_option( 'manual_capture' ) === 'no';
-		$capturable_payment_methods = $capture ? $this->get_upe_enabled_payment_method_ids() : [ 'card' ];
+		$capturable_payment_methods = $capture ? $this->get_upe_enabled_payment_method_ids() : [ 'card', 'link' ];
```
Comment (kalessil) [inline includes/payment-methods/class-upe-payment-gateway.php]: We have constants available for the payment methods see `Payment_Method` for reference.

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 3]
```diff
@@ -924,7 +924,7 @@ function() use ( $payment_fields ) {
 	 */
 	public function get_payment_method_ids_enabled_at_checkout( $order_id = null, $force_currency_check = false ) {
 		$capture                    = empty( $this->get_option( 'manual_capture' ) ) || $this->get_option( 'manual_capture' ) === 'no';
```
Comment (kalessil) [inline includes/payment-methods/class-upe-payment-gateway.php]: While we are touching on this, the variable name is misleading. Can you tweak the name, please?

**Review by kalessil — CHANGES_REQUESTED** [review]
Good catch folks! I like the simplicity of fix and left some comments to address before we can move forward.

**Comment by kalessil** [comment]
Can we also create a PR for updating PM docs to mention auth/capture support?

**Comment by kalessil** [comment]
@dpaun1985, are we ready for the next review round here?

**Comment by dpaun1985** [comment]
> @dpaun1985, are we ready for the next review round here?

yep.

**Review by kalessil — CHANGES_REQUESTED** [review]
Did we consider a testing scenario when UPE is disabled (for both US and non-US merchants)?
If not, please ensure the checkout is functional for new cases and update testing instructions in the PR.

**Comment by dpaun1985** [comment]
> Did we consider a testing scenario when UPE is disabled (for both US and non-US merchants)?

I think Link works only with UPE. I will double-check.

**Comment by kalessil** [comment]
You know, you are right - the change is in UPE gateways. Sorry for the false-alarm, I'll test now.

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 4]
```diff
@@ -923,8 +924,8 @@ function() use ( $payment_fields ) {
 	 * @return string[]
 	 */
 	public function get_payment_method_ids_enabled_at_checkout( $order_id = null, $force_currency_check = false ) {
-		$capture                    = empty( $this->get_option( 'manual_capture' ) ) || $this->get_option( 'manual_capture' ) === 'no';
-		$capturable_payment_methods = $capture ? $this->get_upe_enabled_payment_method_ids() : [ 'card' ];
+		$automatic_capture          = empty( $this->get_option( 'manual_capture' ) ) || $this->get_option( 'manual_capture' ) === 'no';
+		$capturable_payment_methods = $automatic_capture ? $this->get_upe_enabled_payment_method_ids() : [ Payment_Method::CARD, Payment_Method::LINK ];
```
Comment (kalessil) [inline includes/payment-methods/class-upe-payment-gateway.php]: Okay, this way, independently of whether Stripe Link is enabled or not in the settings - the link will be available.

While `capturable_payment_methods` naming is also misleading (please rename it to reflect the reality of what's happening there), I still suggest a pseudo-code to consider:
```php
		$automatic_capture = empty( $this->get_option( 'manual_capture' ) ) || $this->get_option( 'manual_capture' ) === 'no';
		if ( $automatic_capture ) {
			$capturable_payment_methods = $this->get_upe_enabled_payment_method_ids();
		} else {
			$capturable_payment_methods = array_intersect( $this->get_upe_enabled_payment_method_ids(), ['card', 'link'] );
		}
```

My change ensures that the hardcode PMs will be available only if those are enabled.

**Review by kalessil — CHANGES_REQUESTED** [review]
Found a bug - please reference to comment below.

**Review by kalessil — APPROVED** [review]
LGTM!

**Review by mordeth — APPROVED** [review]
Looking good. Tested it, works well and no errors are displayed.

**Pull request merged**